server/src/test/java/io/crate/statistics/ReservoirTest.java

```diff
@@ -40,4 +42,18 @@ public void test_sampling() {
         }
         assertThat(samples.samples(), contains(83, 50, 13, 18, 38));
     }
+
+    @Test
+    public void test_reservoir_is_protected_against_integer_overflow() throws Exception {
+        Random random = new Random(42);
+
+        Reservoir<Long> samples = new Reservoir<>(5, random);
+        Field f1 = samples.getClass().getDeclaredField("itemsSeen");
```

Comment:
I think I'd just make `itemsSeen` package private to avoid the reflection, but no strong opinion.

Reply:
I'd prefer not to add this as it would only be needed for testing.